Core/Utility/DiagnosticsClient.cs

```diff
@@ -56,6 +56,29 @@ public static void TrackException(Exception exception, IDictionary<string, strin
             _client.TrackException(exception, properties, metrics);
         }
 
+        public static void TrackException(Exception exception, IPackage package, bool packageIsPublic, IDictionary<string, string>? properties = null, IDictionary<string, double>? metrics = null)
+        {
+            if (_client == null) return;
+
+
+            if (packageIsPublic && package != null) // Only log public packages
+            {
+                var info = new Dictionary<string, string>
+                    {
+                        { "packageId", package.Id },
+                        { "packageVersion", package.Version.ToNormalizedString() }
+                    };
+
+                TrackException(exception, info);
+            }
+            else
+            {
+                TrackException(exception);
+            }
+
+            _client.TrackException(exception, properties, metrics);
```

Comment:
Do the above `TrackException`s also call `_client.TrackException` so you they would be reported twice?